src/DataMapper/SubscriptionMapper.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GroupLife\Core\DataMapper;
+
+use GroupLife\Core\Exception\LoadFromDbImpossible;
+use GroupLife\Core\Exception\SavingToDbIsForbidden;
+use GroupLife\Core\Subscription\SubscriptionInterface;
+
+class SubscriptionMapper
+{
+    /**
+     * @var \Doctrine\DBAL\Connection
+     */
+    private $connection;
+
+    /**
+     * @param \Doctrine\DBAL\Connection $connection
+     */
+    public function __construct(\Doctrine\DBAL\Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    /**
+     * @param SubscriptionInterface $subscription
+     * @throws SavingToDbIsForbidden
+     * @throws \Doctrine\DBAL\Exception
+     */
+    public function insert(SubscriptionInterface $subscription)
+    {
+        $data = getDataObject($subscription);
+        if (!empty($data->id)) {
+            throw new SavingToDbIsForbidden('Object is already in the database');
+        }
+        if ($data->activity instanceof \stdClass) {
```

Comment:
Why do you have to assert the instance here?

Reply:
Fixed